custom/_legacy/hsph/tasks.py

```diff
@@ -107,9 +107,14 @@ def new_update_case_properties():
             next_assignment = get_none_or_value(case, "next_assignment")
             facility_id = get_none_or_value(case, "facility_id")
             fida_group = get_group_id(domain, "fida", facility_id)
-            cati_owner_username = get_owner_username(domain, "cati", facility_id)
 
-            ## Assignment Directly from Registration ##
+            # get cati_owner_username from current owner-group
+            assigned_owner_group = get_none_or_value(case, "owner_id")
+            if not GROUPS_BY_ID[assigned_owner_group]:
```

Comment:
is this guaranteed to succeed?  What if `assigned_owner_group` is `None`?  Should it be
`if not GROUPS_BY_ID.get(assigned_owner_group, None):`